rest_server/src/models/marketplace_item.js

```diff
@@ -246,6 +246,14 @@ class MarketplaceItem {
 
   async create(itemReq) {
     const handler = modelSyncHandler(async itemReq => {
+      if (itemReq.isPrivate) {
+        itemReq.isPublic = false;
+        itemReq.groupList = [];
+      } else if (itemReq.isPublic) {
+        itemReq.isPrivate = false;
+        itemReq.groupList = [];
```

Comment:
`itemReq.groupList = []` can moved out of the if/else block.

Reply:
this is a little tricky since there may be other cases (`isPublic` and `isPrivate` are same), and we don't know if in those cases whether we should set the groupList as null. 🤣

Reply:
If the item is share to some specific groups, we use `isPrivate: false & isPulbic: false` in the request body, and set the groups into `groupList`. In this case, the groupList should not be empty.
But I think the `itemReq.isPrivate = false;` can be removed. 🤣